docker/build-tools/build-and-push.sh

```diff
@@ -16,9 +16,18 @@
 
 # support other container tools, e.g. podman
 CONTAINER_CLI=${CONTAINER_CLI:-docker}
-HUB=gcr.io/istio-testing
-VERSION=$(date +%Y-%m-%dT%H-%M-%S)
 
-${CONTAINER_CLI} build -t "${HUB}/build-tools:${VERSION}" -t "${HUB}/build-tools:latest" .
+HUB=${HUB:-gcr.io/istio-testing}
+DATE=$(date +%Y-%m-%dT%H-%M-%S)
+BRANCH=$(git branch --show-current 2>/dev/null || echo "master")
+VERSION="${BRANCH}-${DATE}"
+
+${CONTAINER_CLI} build -t "${HUB}/build-tools:${VERSION}" -t "${HUB}/build-tools:${BRANCH}-latest" .
+
 ${CONTAINER_CLI} push "${HUB}/build-tools:${VERSION}"
-${CONTAINER_CLI} push "${HUB}/build-tools:latest"
```

Comment:
This will only show the latest for each branch, if that's what you want. You will see a lot of old images, but not necessarily know which branch they are for. If you want to have the tag persist, you would need to push another image with just the branch tag. It'll then continue to show the branch on the image once the next image is made

Reply:
We also made `VERSION="${BRANCH}-${DATE}"` so for each build we will make `master-20190123` and `master-latest`. Seems right to me? I may be misunderstanding you though

Reply:
OK. You'll always have a single `:master-latest` and a single `:release-1.4-latest` and then various others like `:master-2019-10`, `:master-2019-12`, `:release-1.4-2019-11`, and `:release-1.4-2019-12` (shortening the dates) (and a bunch of other older images) , and that sounds like what you want. I had thought you wanted a slightly different naming structure, but this is fine.